packages/angularjs-auth-passport/lib/mobileAuthService.js

```diff
@@ -57,21 +58,29 @@ MobileAuthService.prototype.authenticate = function(username, password) {
   });
 }
 
-MobileAuthService.prototype.setListener = function(listener) {
+MobileAuthService.prototype.setLoginListener = function(listener) {
   this.loginListener = listener;
 }
 
+MobileAuthService.prototype.setLogoutListener = function(listener) {
+  this.logoutListener = listener;
+}
+
 MobileAuthService.prototype.login = function() {
   var self = this;
+  if (self.logoutListener) {
+    self.logoutListener();
+  }
+  localStorage.clear();
   return self.state.go(CONSTANTS.LOGIN_STATE_ROUTE);
 };
 
 MobileAuthService.prototype.logout = function() {
   var self = this;
-  localStorage.clear();
-  if (self.loginListener) {
-    self.loginListener();
+  if (self.logoutListener) {
```

Comment:
I do not really see any difference between login and logout handlers. 
In fact in code when we mean logout we actually mean to call login. 
With that in mind can we call login from logout method so it will be lss confusing